src/modules/fancyzones/lib/JsonHelpers.cpp

```diff
@@ -0,0 +1,965 @@
+    {
+        try
+        {
+            auto devices = fancyZonesDataJSON.GetNamedArray(L"devices");
+
+            for (int i = 0; i < devices.Size(); ++i)
+            {
+                const auto device = DeviceInfoJSON::DeviceInfoJSON::FromJson(devices.GetObjectAt(i));
+                if (device.has_value())
+                {
+                    deviceInfoMap[device->deviceId] = device->data;
+                }
+                else
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+        catch (const winrt::hresult_error&)
+        {
+            return false;
+        }
+    }
+
+    json::JsonArray FancyZonesData::SerializeDeviceInfos() const
+    {
+        json::JsonArray DeviceInfosJSON{};
+
+        int i = 0;
+        for (const auto& [deviceID, deviceData] : deviceInfoMap)
+        {
+            DeviceInfosJSON.Append(DeviceInfoJSON::DeviceInfoJSON::ToJson(DeviceInfoJSON{ deviceID, deviceData }));
+        }
+
+        return DeviceInfosJSON;
+    }
+
+    bool FancyZonesData::ParseCustomZoneSets(const json::JsonObject& fancyZonesDataJSON)
+    {
+        try
+        {
+            auto customZoneSets = fancyZonesDataJSON.GetNamedArray(L"custom-zone-sets");
+
+            for (int i = 0; i < customZoneSets.Size(); ++i)
+            {
+                const auto zoneSet = CustomZoneSetJSON::FromJson(customZoneSets.GetObjectAt(i));
+                if (zoneSet.has_value())
+                {
+                    customZoneSetsMap[zoneSet->uuid] = zoneSet->data;
+                }
+            }
+
+            return true;
+        }
+        catch (const winrt::hresult_error& e)
+        {
+            return false;
+        }
+    }
+
+    json::JsonArray FancyZonesData::SerializeCustomZoneSets() const
+    {
+        json::JsonArray customZoneSetsJSON{};
+
+        int i = 0;
+        for (const auto& [zoneSetId, zoneSetData] : customZoneSetsMap)
+        {
+            customZoneSetsJSON.Append(CustomZoneSetJSON::ToJson(CustomZoneSetJSON{ zoneSetId, zoneSetData }));
+        }
+
+        return customZoneSetsJSON;
+    }
+
+    void FancyZonesData::CustomZoneSetsToJsonFile(const std::wstring& filePath) const
+    {
+        const auto& customZoneSetsJson = SerializeCustomZoneSets();
+        json::JsonObject root{};
+        root.SetNamedValue(L"custom-zone-sets", customZoneSetsJson);
+        json::to_file(filePath, root);
+    }
+
+    void FancyZonesData::LoadFancyZonesData()
+    {
+        std::wstring jsonFilePath = GetPersistFancyZonesJSONPath();
+
+        if (!std::filesystem::exists(jsonFilePath))
+        {
+            TmpMigrateAppliedZoneSetsFromRegistry();
+
+            // Custom zone sets have to be migrated before applied zone sets!
+            MigrateCustomZoneSetsFromRegistry();
+            MigrateAppZoneHistoryFromRegistry();
+            //MigrateDeviceInfoFromRegistry(appliedZoneSets);
```

Comment:
Why this is commented out?

Reply:
Devices info is migrated when editor is spawned on that specific device (ZoneWindow.cpp -> LoadSettings() ).

When this wasn't commented, the logic was to go through all devices in registries and migrate all of them, but there could be devices that are no longer present.. Also, device uniqueId has virtualDesktopId which could have changed in the meantime..